contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+        uint256 _usdtBalance = IERC20(_usdt).balanceOf(address(this));
+        uint256 _3crvBalanceBefore = IERC20(_3crv).balanceOf(address(this));
+        if (_daiBalance != 0 || _usdcBalance != 0 || _usdtBalance != 0) {
+            uint256[3] memory _add3poolLiquidityAmounts;
+            _add3poolLiquidityAmounts[0] = _daiBalance;
+            _add3poolLiquidityAmounts[1] = _usdcBalance;
+            _add3poolLiquidityAmounts[2] = _usdtBalance;
+            IERC20(_dai).approve(_3pool, _daiBalance);
+            IERC20(_usdc).approve(_3pool, _usdcBalance);
+            SafeERC20.safeApprove(IERC20(_usdt), _3pool, _usdtBalance);
+            I3Pool(_3pool).add_liquidity(
+              _add3poolLiquidityAmounts,
+              0 // minimum 0 3crv lp tokens out (defensively checked below)
+            );
+        }
+
+        // get the number of 3crv held by the contract, and also
+        // check for slippage during the 3pool deposit
+        uint256 _3crvBalanceAfter = IERC20(_3crv).balanceOf(address(this));
+        uint256 _3crvBalanceFromDeposit = _3crvBalanceAfter - _3crvBalanceBefore;
+        uint256 _min3crvOut = (_daiBalance + _usdcBalance + _usdtBalance) * 1e18 / _3crvVirtualPrice * (BASIS_POINTS_GRANULARITY - depositMaxSlippageBasisPoints) / BASIS_POINTS_GRANULARITY;
+        require(_3crvBalanceFromDeposit >= _min3crvOut, "StableSwwapOperatorV1: 3pool deposit slippage too high");
+
+        // get the amount of tokens in the pool
+        (uint256 _3crvAmount, uint256 _feiAmount) = (
+            IStableSwap(pool).balances(_3crvIndex),
+            IStableSwap(pool).balances(_feiIndex)
+        );
+        // ... and the expected amount of 3crv in it after deposit
+        uint256 _3crvAmountAfter = _3crvAmount + _3crvBalanceAfter;
+
+        // get the usd value of 3crv in the pool
+        uint256 _3crvUsdValue = _3crvAmountAfter * _3crvVirtualPrice / 1e18;
+
+        // compute the number of FEI to deposit
+        uint256 _feiToDeposit = 0;
+        if (_3crvUsdValue > _feiAmount) {
+            _feiToDeposit = _3crvUsdValue - _feiAmount;
+        }
+
+        // deposit
+        if (_3crvBalanceAfter > 0 || _feiToDeposit > 0) {
+            // build parameters
+            uint256[2] memory _addLiquidityAmounts;
+            _addLiquidityAmounts[_feiIndex] = _feiToDeposit;
+            _addLiquidityAmounts[_3crvIndex] = _3crvBalanceAfter;
+
+            // approvals
+            IERC20(address(_fei)).approve(pool, _feiToDeposit);
+            // 3crv needs to reset allowance to 0, see CurveTokenV1.vy#L117 :
+            // assert _value == 0 or self.allowances[msg.sender][_spender] == 0
+            IERC20(_3crv).approve(pool, 0);
+            IERC20(_3crv).approve(pool, _3crvBalanceAfter);
+
+            // do deposit
+            uint256 _balanceBefore = IERC20(pool).balanceOf(address(this));
+            IStableSwap(pool).add_liquidity(_addLiquidityAmounts, 0);
```

Comment:
No slippage protection here?

Reply:
I don't think it's needed, because we deposit in proportions that makes the pool balanced, unlike when we deposit/withdraw 100% DAI (instead of a basket of DAI/USDC/USDT), where slippage can happen ?

Reply:
Doesn't ever hurt

Reply:
Added a slippage check there